src/components/send-message/SendMessage.js

```diff
@@ -1,15 +1,21 @@
 import React, { useState } from "react";
+
 import "./SendMessage.css";
+
 import question from "../../images/question.png";
-function SendMessage() {
+import { sendMessage } from "../../utils/Contract";
+
+function SendMessage(props) {
+  const { walletProvider } = props;
+
   const [recipientWallet, setRecipientWallet] = useState("");
   const [message, setMessage] = useState("");
   const [email, setEmail] = useState("");
   const [bidAmount, setBidAmount] = useState("");
 
-  const sendMessage = (e) => {
+  const send = (e) => {
     e.preventDefault();
-    console.log(" all", recipientWallet, message, email, bidAmount);
+    sendMessage(walletProvider, message, recipientWallet, email, bidAmount);
```

Comment:
My bad for changing these parameters around on you.

Reply:
all good!